contracts/FIFOMarket.sol

```diff
@@ -136,9 +155,14 @@ contract FIFOMarket is
   ) public override returns (bytes4) {
     for (uint256 i = 0; i < ids.length; i++) {
       uint256 removalAmount = _removal.balanceOf(address(this), ids[i]);
-      totalSupply += removalAmount;
-      _queue[_queueNextInsertIndex] = (ids[i]);
-      _queueNextInsertIndex++;
+      totalActiveSupply += removalAmount;
+      totalNumberActiveRemovals += 1;
+      address supplierAddress = ids[i].supplierAddress();
+      _activeSupply[supplierAddress].add(ids[i]);
+      if (!_activeSuppliersUnordered[supplierAddress]) {
+        _activeSuppliersUnordered[supplierAddress] = true;
+        _activeSuppliersOrdered.push(supplierAddress);
+      }
```

Comment:
if this supplier doesn't yet or currently have any active supply, we have to add them to the collections we use to keep track of the active sellers